### ISSUE TITLE
Unable to pass variable into contains condition - use this instead for this edge case

### ISSUE THREAD

**Issue by Whitesushi** [kelaberetiv/TagUI#1014]
I'm trying to loop through an array to perform an action if a keyword is detected within the array. I'm trying to use the "contains" functions but if I write the variable inside directly, I get error saying "no quoted text" and if I were to wrap it in ` `, it doesn't work even though when I try to print the result, it matches the string. I double checked to make sure the function itself works by inputting a string like "tail" instead of a variable and it does. I also combed through the files looking for any instance mention of contains but all the examples seem to be for strings 

Here's an example

```
keywords_array = ["one tails","two heads","three coins"]
keyword = "tail"

	for (n=0; n<keywords_array.length; n++)
		if keywords_array[n] contains keyword
			echo hello
```

**Comment by kensoh** [comment]
Hi @Whitesushi this is by design to help users detect mistake in using conditions and not providing a string to compare with `contains`. But I definitely understand why this error detection will end up causing problem for your use case when comparing with a variable instead. Let me look into that and see what is the best way to handle, and see if codebase can be improved to handle this.

Thanks for your feedback! I'll raise this as a bug because although intention is good, it causes problem for this case.

**Comment by ruthtxh** [comment]
Hi @Whitesushi 

Meanwhile you can write some vanilla javascript code to compare... something like this. Hope this helps!

```
keywords_array = ["one tails","two heads","three coins"]
arr = []
keyword = "tail"
js begin
for (n=0; n<keywords_array.length; n++)
	if (keywords_array[n].indexOf(keyword) !== -1)
		arr.push(true);
	else arr.push(false);
js finish
echo `arr`	
```

**Comment by Whitesushi** [comment]
Hi @ruthtxh 

With some modifications I was able to implement the workaround into my code and it works, albeit slow since the slight delay of doing an additional check adds up when I'm looping through large amounts of data. Either way, thanks a lot for the solution!

**Comment by kensoh** [comment]
Oh great! I'll be continuing to explore a permanent solution for this. For Ruth's solution, it shouldn't add on to execution time because TagUI conditions 'compiles' to JavaScript too and run as JavaScript code.

However when for loop is written in human language (which is the default expected usage) `for n from 1 to 20`, this will really have significant delay as the range goes up. This is because in this case, a complete stack data structure will be built before the loop starts.

**Comment by kensoh** [comment]
Hi @Whitesushi and @ruthtxh I've looked again and it looks like it is hard to change this from codebase because for the primary use case of contains condition it uses the quotes to know how to convert into correct JavaScript code. Without the quotes it is hard to correctly fine the variable to be compared to.

What I've found is an easy workaround, by comparing against '\`variable\`'. See working example below -

```
text = 'abc'
q = 'a'

if text contains '`q`'
    echo YES

if text contains 'b'
    echo YES
```

```
START - automation started - Fri May 21 2021 11:26:48 GMT+0800 (+08)

YES
YES

FINISH - automation finished - 0.2s
```

I'll close this issue but will look out for more inputs, if any, from you!
